userspace/chisel/lua_parser.cpp

```diff
@@ -39,50 +38,29 @@ const static struct luaL_reg ll_filter [] =
 	{NULL,NULL}
 };
 
-lua_parser::lua_parser(gen_event_filter_factory &factory, lua_State *ls, const char *lua_library_name)
-	: m_factory(factory)
+lua_parser::lua_parser(std::shared_ptr<gen_event_filter_factory> factory)
+	: m_factory(factory), m_last_boolop(BO_NONE),
+	  m_have_rel_expr(false), m_nest_level(0)
 {
-	m_filter = NULL;
-
-	m_ls = ls;
-	reset();
-
-	// Register our c++ defined functions
-	luaL_openlib(m_ls, lua_library_name, ll_filter, 0);
+	m_filter.reset(m_factory->new_filter());
```

Comment:
This can't be done in the initializer list I expect?

Reply:
No, it can. I'll change it.